### PR TITLE
Add explicit error message when creating/updating image

### PR DESCRIPTION
**Pull request by mgold1234**
realted to THEEDGE-2305

Signed-off-by: mgold1234 <mgold@redhat.com>

# Description
Add explicit error message when creating/updating image with already existing image name or image set
Fixes # (issue)

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [x] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
a2e875d8-c617-4a1e-ac43-a409178756f5

File: pkg/routes/images_test.go [hunk 1]
```diff
@@ -129,6 +130,150 @@ func TestCreate(t *testing.T) {
...
+	req, err := http.NewRequest("POST", "/", &buf)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := req.Context()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockImageService := mock_services.NewMockImageServiceInterface(ctrl)
+	mockImageService.EXPECT().CreateImage(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(new(services.PackageNameDoesNotExist))
+	if err != nil {
```
Comment (ldjebran) [inline pkg/routes/images_test.go:163]: delete this block

File: pkg/routes/images_test.go [hunk 2]
```diff
@@ -129,6 +130,150 @@ func TestCreate(t *testing.T) {
...
+	req, err := http.NewRequest("POST", "/", &buf)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := req.Context()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockImageService := mock_services.NewMockImageServiceInterface(ctrl)
+	mockImageService.EXPECT().CreateImage(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(new(services.ThirdPartyRepositoryInfoIsInvalid))
+	if err != nil {
```
Comment (ldjebran) [inline pkg/routes/images_test.go:211]: remove this block

File: pkg/routes/images_test.go [hunk 3]
```diff
@@ -129,6 +130,150 @@ func TestCreate(t *testing.T) {
...
+	req, err := http.NewRequest("POST", "/", &buf)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx := req.Context()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockImageService := mock_services.NewMockImageServiceInterface(ctrl)
+	mockImageService.EXPECT().CreateImage(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(new(services.ImageNameAlreadyExists))
+	if err != nil {
```
Comment (ldjebran) [inline pkg/routes/images_test.go:259]: remove this block

**Review by ldjebran — APPROVED** [review]
LGTM

**Pull request merged**